index.d.ts

```diff
@@ -0,0 +1,112 @@
+/** Changes the return type of a function to `any`. */
+type ReturningAny<T extends (...args: any[]) => any> = (...a: Parameters<T>) => any;
+
+/** @see {@link https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/globals.d.ts|Process.on()} */
+type ProcessEventOrSignal =
+  NodeJS.Signals
+  | 'beforeExit'
+  | 'disconnect'
+  | 'exit'
+  | 'rejectionHandled'
+  | 'uncaughtException'
+  | 'unhandledRejection'
+  | 'warning'
+  | 'message'
+  | 'newListener'
+  | 'removeListener'
+  | 'multipleResolves';
+
+declare namespace AsyncExitHook {
+
+  /** The callback to call when the event handling is finished. */
+  type AsycHookCallback = () => void;
```

Comment:
Nit: typo here: should be `AsyncHookCallback`. Use sites will also need to change.